include/net/net_if.h

```diff
@@ -454,7 +454,7 @@ struct net_if_dev {
 
 #if defined(CONFIG_NET_SOCKETS_OFFLOAD)
 	/** Indicate whether interface is offloaded at socket level. */
-	bool offloaded;
+	int (*socket_offload)(int family, int type, int proto);
```

Comment:
Comment needs to be updated. Also the name `socket_offload` does not suggest that this is for socket creation, so maybe a more descriptive name could be used.

Reply:
Updated the comment. I'm not sure about the naming, IMO `socket_offload` is quite straightforward (just like regular `socket()` but with a suffix indicating it refers to an offloaded interface). Do you have any suggestions regarding an alternative name?

Reply:
`socket()` is known enough to get idea what it does (socket creation). It was not that clear to me when seeing `->socket_offload(family, type, proto);` callback invocation. Was thinking about `socket_offload_create`, but not sure that improves something.